app/src/modules/tageuro/tageuro.js

```diff
@@ -0,0 +1,191 @@
+     *  values would be like this :
+     *  	* values[0] = "test working?"
+     *  	* values[1] = "test"
+     *  	* values[2] = "working?"
+     *  But, if you use the trigger "strict", there will be only one value in this array (values[0] = "test working?")
+     *  - proxyURL : If you need to call an external API, use the package "request" with proxyURL as value for
+     *  "proxy" attribute (you can refer to weather IA to check how to use it properly)
+     *  - language : Current language called
+     * @returns data to be displayed.
+     */
+
+    getData: function (values, proxyURL, language, i18n) {
+        const _ = i18n._;
+        return new Promise(function (resolve, reject) {
+
+            if (values[0]) {
+                const CACHE_KEY = 'tageuro-cache';
+                const CACHE_EXPIRE = 10;
+                const redisTools = require('../../redis_tools');
+                redisTools.initRedis();
+                             
+                // Checking the cache to avoid requesting the API
+                redisTools.getFromCache(CACHE_KEY).then((cached) => {
+                    if (cached) { // we already have a recent answer
+                        cached = JSON.parse(cached);
+                        if (cached) {
+                            cached.fromCache = true;
+                            resolve(cached);
+                        } else {
+                            reject("Error: something went wrong while fetching cached response");
+                        }
+                    } else { // no cache
+                    
+                        // Declares the API Caller
+                        var apiCaller = require('../../api_caller');
+
+                        var result = encodeURI(values[0]);
+                        var api_request = 'https://www.sentimantic.net/qwant/apiReqIA/image?fid='+result;
+
+                        // Defines the structure of the answer
+                        var structure = {
+                            "image":  "String"
+                        };
+                            
+
+                        // Call the API and get back data
+                        apiCaller.call(api_request, structure, proxyURL, this.timeout, redisTools).then((apiRes) => {
+                             redisTools.saveToCache(CACHE_KEY, apiRes, CACHE_EXPIRE);
+                            apiRes.fromCache = false;
+                            resolve(apiRes);
+                        }).catch((error) => {
+                            reject(error);
+                        });
+                            
+                    }
+                });
+            } else {
+                reject("Couldn't process query.")
+            }
+        });
+    },
+
+    /**
+     * (OPTIONAL/NEEDED)
+     * If your name needs to be translated, use this function getName().
+     * @returns the tab name translated
+     */
+
+    getName: function (i18n) {
```

Comment:
The function name doesn't need to be translated here, so the "name" field is sufficient
Please delete this function